dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+  of the observer.
+
+- It should be convenient to do timezone conversions easily.
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification
+#
+# Because there may be many attributes for an observer, I suggest using
+# keyword parameters for most of them.
+from astroplan.entity import Observer
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
+               latitude='+19:49:42.600',
+               elevation=4163,
+               pressure=615,
+               relative_humidity=0.11,
+               temperature=0,
+               timezone='US/Hawaii',
+               description="Subaru Telescope on Mauna Kea, Hawaii")
+
+# longitude and latitude would be represented internally by astropy
+# angles.Longitude/Latitude.  So any value that can naturally initialize
+# one of those objects could be passed as the parameter to Observer
+# Other items are represented internally using the appropriate astropy
+# Quantities and Units and so can be initialized similarly.
+
+# It's also possible to initialize using astropy objects and other
+# direct entities
+from astropy.coordinates.angles import Longitude, Latitude
+import astropy.units as u
+import pytz
+
+obs = entity.Observer(name='Keck 1',
+                      longitude=Longitude(155.4750 * u.degree),
+                      latitude=Latitude(19.8264 * u.degree)
+                      elevation=4160 * u.m,
+                      pressure=615 * u.hPa,
+                      relative_humidity=0.11,
+                      temperature=0.0 * u.C,
+                      timezone=pytz.timezone('US/Hawaii'),
+                      description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.get_observer('keck1')
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+#
+obs.set_env_conditions(pressure=600 * u.hPa, relative_humidity=0.2,
+                       temperature=10 * u.C)
+
+# It's easy to construct date/times from the observer, since it knows about
+# its timezone.  These dates can be passed directly to target check methods.
+
+# dates default to midnight if not otherwise specified
+day = obs.get_date("2015-05-01")
```

Comment:
@eteq, the point of this method is to provide a _convenient way_ to get a timezone aware date or datetime in the observing position.  This is trivial if you are _programming_ on top of of this api, but for end users simply planning an observation--if we don't provide a convenience method, then we force every user to write over and over the one or two line function to convert a time expressed in hours and minutes to the local observing time.  I'm aware that some observers plan the observation is UTC, but I've seen many that plan in the local time, and prefer the airmass plots in the local time.